src/cascadia/TerminalSettingsEditor/Appearances.xaml

```diff
@@ -183,44 +183,39 @@
-                    <!--
-                        Binding the ItemsSource to a separate variable that switches between the
-                        two font lists causes a crash within the ComboBox code.
-                        As a workaround, introduce two ComboBox controls and only display one at a time.
-                    -->
-                    <ComboBox x:Uid="Profile_FontFaceBox"
-                              ItemTemplate="{StaticResource FontFaceComboBoxItemTemplate}"
-                              ItemsSource="{x:Bind SourceProfile.MonospaceFontList, Mode=OneWay}"
-                              SelectedItem="{x:Bind CurrentFontFace, Mode=OneWay}"
-                              SelectionChanged="FontFace_SelectionChanged"
-                              Style="{StaticResource ComboBoxSettingStyle}"
-                              Visibility="{x:Bind mtu:Converters.InvertedBooleanToVisibility(ShowAllFonts), Mode=OneWay}" />
-                    <ComboBox x:Uid="Profile_FontFaceBox"
-                              ItemTemplate="{StaticResource FontFaceComboBoxItemTemplate}"
-                              ItemsSource="{x:Bind SourceProfile.CompleteFontList, Mode=OneWay}"
-                              SelectedItem="{x:Bind CurrentFontFace, Mode=OneWay}"
-                              SelectionChanged="FontFace_SelectionChanged"
-                              Style="{StaticResource ComboBoxSettingStyle}"
-                              Visibility="{x:Bind ShowAllFonts, Mode=OneWay}" />
+                    <AutoSuggestBox x:Name="_fontFaceBox"
+                                    x:Uid="Profile_FontFaceBox"
+                                    GotFocus="FontFaceBox_GotFocus"
+                                    ItemTemplate="{StaticResource FontFaceComboBoxItemTemplate}"
+                                    ItemsSource="{x:Bind FilteredFontList, Mode=OneWay}"
+                                    LostFocus="FontFaceBox_LostFocus"
+                                    SuggestionChosen="FontFaceBox_SuggestionChosen"
+                                    Text="{x:Bind Appearance.FontFace, Mode=OneWay}"
+                                    TextBoxStyle="{StaticResource TextBoxSettingStyle}"
+                                    TextChanged="FontFaceBox_TextChanged" />
                     <CheckBox x:Name="ShowAllFontsCheckbox"
                               x:Uid="Profile_FontFaceShowAllFonts"
-                              IsChecked="{x:Bind ShowAllFonts, Mode=TwoWay}"
-                              IsEnabled="{x:Bind UsingMonospaceFont, Mode=OneWay}" />
+                              IsChecked="{x:Bind ShowAllFonts, Mode=TwoWay}" />
                 </StackPanel>
             </local:SettingContainer>
-            <muxc:InfoBar x:Uid="Profile_FontFace_ProportionalFontWarning"
-                          IsOpen="{x:Bind ShowProportionalFontWarning, Mode=OneWay}"
+            <muxc:InfoBar x:Uid="Profile_MissingFontFaces"

```

Comment:
OKAY SO.
If you want to use the new container styles I created for this.

```xml
<local:SettingContainer x:Uid="Profile_MissingFontFaces" Style="{StaticResource SettingContainerWarningStyle}" />
```

You can set the *HelpText* instead of the `Message`, and you need the resw file to set the *Header* instead of the `Title`. Visibility instead of `IsOpen`.

Now that resources are checked in, you can also delete the old keys for the ones you removed.

I recommend still using *new keys* however, because there's a nasty issue with Uids in multiple languages which will destroy your life.

<details>

<summary>resw uid issue</summary>

If the en-US resw contains `Foo_Bar.Header`, but all the language resources contain `Foo_Bar.Title`, the resource compiler will throw a warning during build. And then your application will crash instantly when `Foo_Bar` shows up on the screen, as it can't apply all the properties since they don't exist in all languages or there's a discordance between them.

</details>

Reply:
The other one is SettingContainerErrorStyle.